src/components/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,140 @@
+import {
+  ChangeEvent,
+  ComponentProps,
+  FocusEvent,
+  MouseEvent,
+  ReactNode,
+  useEffect,
+  useMemo,
+  useState,
+} from "react"
+import { useKeyPress } from "../../hooks/useKeyPress"
+import styles from "./SearchInput.module.css"
+
+const DEFAULT_DELAY = 300
+
+type Omitted = "onChange" | "type" | "value"
+
+interface SearchInputProps extends Omit<ComponentProps<"input">, Omitted> {
+  "aria-label": string
+  searchIcon?: ReactNode
+  clearIcon?: ReactNode
+  debounceDelay?: number
+  initialValue?: string
+  onSearch: (searchTerm: string) => void
+  showSearchButton?: boolean
+}
+
+export function SearchInput({
+  searchIcon = <span>🔍</span>,
+  clearIcon = <span>❎</span>,
+  onSearch,
+  debounceDelay = DEFAULT_DELAY,
+  showSearchButton,
+  initialValue = "",
+  ...inputProps
+}: SearchInputProps) {
+  const [searchTerm, setSearchTerm] = useState("")
+  const [clearButtonHidden, setClearButtonHidden] = useState(true)
+
+  const debouncedOnSearch = useMemo(() => {
+    return debounce(onSearch, debounceDelay)
+  }, [debounceDelay, onSearch])
+
+  useKeyPress(["Enter"], () => {
+    debouncedOnSearch(searchTerm)
+  })
+
+  useEffect(() => {
+    setSearchTerm(initialValue)
+  }, [initialValue])
```

Comment:
Beware that if the parent changes the `initialValue` on subsequent renders, it will impact the value of `searchTerm`. So it goes beyond the initial value. Which might be fine depending on how you want this component to behave.

Formik (and other libraries, like redux-form) allow both modes of operation in a similar setup with the boolean `enableReinitialize` attribute. If `true`, changes to the `initialValue` affect the state of the component, if `false`, only the initial value is used, and subsequent changes don't affect the state. You def don't need to do what they did though.